optimized.rb

```diff
@@ -0,0 +1,20 @@
+counts = Hash.new(0)
+bufsize = 64*1024
+remaining = ''
+while 1 do
+  chunk = STDIN.read(bufsize)
+  if !chunk
+    break
+  end
+  chunk = remaining + chunk
+  last_lf = chunk.rindex("\n")
+  if !last_lf
+    remaining = ''
+  else
+    remaining = chunk[last_lf+1..]
+    chunk = chunk[..last_lf]
+  end
+  chunk.downcase.split.each{ |w| counts[w] += 1 }
```

Comment:
I struggle a little about this block invocation with `#each`. It should be more optimized without a block being instantiated but I dont know if this is possible.